[Jonathan] iP

fullfatwasabi opened the pull request:
### SaDOS

> "Welcome to test chamber four." - GLaDOS

Introducing your very own *sad* task manager. It's

* text-based
* easy to learn
* **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/fullfatwasabi/ip/releases/tag/v0.1).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here's an excerpt of the main method.
```java
public static void main(String[] args) {
        Sados currentlist = new Sados();
        currentlist.run();
    }
```

Commit:
Add Gradle support

fullfatwasabi committed:
Level-1

fullfatwasabi committed:
Level-2

fullfatwasabi committed:
Level-1

fullfatwasabi committed:
Level-2

fullfatwasabi committed:
Level-3

fullfatwasabi committed:
Level-4

fullfatwasabi committed:
Level-4

fullfatwasabi committed:
A-TextUiTesting

fullfatwasabi committed:
Handle Errors

fullfatwasabi committed:
Level-5

fullfatwasabi committed:
A-TextUiTesting

fullfatwasabi committed:
Level-5

fullfatwasabi committed:
Level-6

fullfatwasabi committed:
small-bug-fix

fullfatwasabi committed:
Level-6

fullfatwasabi committed:
Level-7 Adding save function

Todo list data is lost after closing, which is troublesome for users.

Add an autosave function that saves everytime user makes changes to the list

fullfatwasabi committed:
Level-8 Adding time functionality and processing

The time can be saved in a more useful manner.

Add an more intuitive processing and saving of the time data field. Add ability to query a specific date.

fullfatwasabi committed:
-

fullfatwasabi committed:
Merged Level-7

fullfatwasabi committed:
-

fullfatwasabi committed:
A-MoreOOP

fullfatwasabi committed:
A-MoreOOP

We face problem of not enough oop, please add more

Adding more oop yay

fullfatwasabi committed:
A-MoreOOP

We face problem of not enough oop, please add more

Adding more oop yay

fullfatwasabi committed:
A-MoreOOP

We face problem of not enough oop, please add more

Adding more oop yay

fullfatwasabi committed:
A-JUnit

We face problem cannot test

Adding more test

fullfatwasabi committed:
A-JavaDoc

Javadoc good, read easy

Adding more javadoc

fullfatwasabi committed:
A-CodingStandard

Standard low bad

Now standard high

fullfatwasabi committed:
-

fullfatwasabi committed:
Level-9

Searching by keyword good

Adding function to search by keyword

fullfatwasabi committed:
Merge branch 'branch-A-JavaDoc'

fullfatwasabi committed:
Merge-A-CodingStandard

fullfatwasabi committed:
Merge-Level-9

fullfatwasabi committed:
-

fullfatwasabi committed:
Rename Test class and methods

fullfatwasabi committed:
Rename main class

fullfatwasabi committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

fullfatwasabi committed:
Add gradle support

Some tasks are tedious and it is not ideal to do them individually.

Gradle would be useful to automate some tasks.

fullfatwasabi committed:
Renamed main class

fullfatwasabi committed:
Add checkstyle setup to gradle

Checkstyle can be quite troublesome and we might miss styling errors.

Automated checkstyle checking would be helpful in saving our time.

fullfatwasabi committed:
no message

fullfatwasabi committed:
Merge branch-A-Gradle and branch-A-Checkstyle to Master

Yukun99 commented on src/main/java/Sados.java:
```diff
@@ -0,0 +1,64 @@
+ */
+
+public class Sados {
+    private Parser parse;
```
Naming convention: parser would be better (nouns for fields)

Yukun99 commented on src/main/java/Sados.java:
```diff
@@ -0,0 +1,64 @@
+            }
+            parse = new Parser(tasks);
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
```
It might be better to print out a more detailed/informative error

Yukun99 commented on src/main/java/Sados.java:
```diff
@@ -0,0 +1,64 @@
+    public void run() {
+        boolean away = false;
+        System.out.println(Ui.BAR + "\n    Hello! I'm SaDOS\n" + "    What can I do for you?\n"
+                + "    Send \"bye\" to exit,\n" + "    I won't hold it against you\n" + "    Send \"help\" for help!\n"
```
\n might not work on some systems, System.lineSeparator() might be good. (I found this after googling it myself)

Yukun99 commented on src/main/java/Sados.java:
```diff
@@ -0,0 +1,64 @@
+                + Ui.BAR);
+        Scanner sc = new Scanner(System.in);
+        do {
+            String str = sc.nextLine();
```
`while (sc.hasNext())` could be a consideration here

Yukun99 commented on src/main/java/Sados.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public static void main(String[] args) {
+        Sados currentlist = new Sados();
```
`lowerCamelCase` for naming fields

Yukun99 commented on src/main/java/addon/Parser.java:
```diff
@@ -0,0 +1,121 @@
+            tasklist.listEntries();
+        } else {
+            if (input.indexOf(' ') > 0) { // checking if command given was >= two words
+                String[] splitted = input.split(" ", 2);
```
`split` is the past tense of split so no need for splitted perhaps

Yukun99 commented on src/main/java/addon/Storage.java:
```diff
@@ -0,0 +1,123 @@
+     * @param list List to be written over.
+     */
+    public static void rewriteFile(ArrayList<Task> list) {
+        File oldFile = new File("./save.txt");
```
Might be good to allow users/developers to select their own filepath as an input

Yukun99 commented on src/main/java/addon/Storage.java:
```diff
@@ -0,0 +1,123 @@
+        Scanner s = new Scanner(saveFile);
+        ArrayList<Task> list = new ArrayList<>();
+
+        while(s.hasNext()) {
```
Could be better to split the parser to a separate method for easier tracking

Yukun99 commented on src/main/java/addon/Storage.java:
```diff
@@ -0,0 +1,123 @@
+                String base = s.nextLine();
+                String[] basesplit = base.split("T", 2);
+                String[] datesplit = basesplit[0].split("-", 3);
+                String[] timesplit = basesplit[1].split(":", 2);
```
Don't forget to use `camelCase` for naming i.e. `baseSplit` or `splitBase`

Yukun99 commented on src/main/java/addon/Storage.java:
```diff
@@ -0,0 +1,123 @@
+                String[] basesplit = base.split("T", 2);
+                String[] datesplit = basesplit[0].split("-", 3);
+                String[] timesplit = basesplit[1].split(":", 2);
+                LocalDateTime sendThis = LocalDateTime.of(Integer.parseInt(datesplit[0]),
```
Naming does not make sense in the context for me

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+     * @param num Index number of the item to be changed.
+     */
+    public void changeDone(int num) throws IncorrectFormatException {
+        if (num >= 1 && num <= this.list.size() + 1) {
```
try-catching a IndexOutOfBoundsException could work too. That way you can possibly print out the error as well for more information.

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * Adds Task entry to the Arraylist
```
Missing period here

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+    /**
+     * Lists out the current entries in the Arraylist.
+     */
+    public void listEntries() {
```
Might be better to abstract all message sending to the Ui class instead of just using it to store constants?

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+                }
+            } else if (i instanceof Deadline) {
+                Deadline o = (Deadline)i;
+                if (o.date.getYear() == date.getYear() && o.date.getMonth().equals(date.getMonth())
+                        && o.date.getDayOfMonth() == date.getDayOfMonth()) {
```
Could be easier to use the .equals() method from the respective classes

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+    /**
+     * Lists out the current entries in the Arraylist that contain the specified keyword.
+     *
+     * @param keyword Keyword to be queried
```
Missing period here

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * Parent class Task, contains doneness and task name
```
Missing period here

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+
+        protected LocalDateTime date;
+
+        public Deadline(String description, LocalDateTime by) {
```
Missing JavaDoc for constructor

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * Child class of Task
```
Missing period and more detailed information about what type of task

Yukun99 commented on src/main/java/addon/Tasklist.java:
```diff
@@ -0,0 +1,226 @@
+
+        protected LocalDateTime date;
+
+        public Event(String description, LocalDateTime at) {
```
Missing JavaDoc for constructor

Yukun99 commented on src/test/TasklistTest.java:
```diff
@@ -0,0 +1,33 @@
+                8,24,0,0));
+        assertEquals(testDeadline.toString(), "[D][ ] homework (by: AUG 24 2021 @ 0000)");
+    }
+    @Test
```
A line between each method would be nice

Yukun99 reviewed (commented):
Everything is generally fine, but some public methods are missing JavaDocs. Maybe you can try more abstraction for certain parts of your program to segregate the parts further!

fullfatwasabi committed:
Add GUI

Text based UI might not be the most user friendly.

Lets add a nicer UI

fullfatwasabi committed:
Add GUI

Text based UI might not be the most user friendly.

Lets add a nicer UI

myathk commented on src/main/java/addon/Storage.java:
```diff
@@ -0,0 +1,84 @@
+        ArrayList<Task> list = new ArrayList<>();
+
+        while (s.hasNext()) {
+            String type = s.nextLine();
+            String name = s.nextLine();
```
Perhaps naming these variables as taskType and taskName will make them more detailed and accurate?

myathk commented on src/main/java/addon/Task.java:
```diff
@@ -0,0 +1,132 @@
+            return "[D]" + super.toString() + " (by: " + Ui.printDate(date) + ")";
+        }
+
+        public boolean dateEquals(LocalDate theirDate) {
```
Should the method be named isDateEquals instead to better describe it as a boolean?

myathk commented on src/main/java/addon/Task.java:
```diff
@@ -0,0 +1,132 @@
+     * @param str String to be queried.
+     * @return true if found, false otherwise.
+     */
+    public boolean nameContains(String str) {
```
Alternatively, maybe the method name "doesNameContain" seems more appropriate for a boolean?

myathk reviewed (commented):
Overall, the code follows proper naming of classes/variables and methods/functions very closely. Just a few nits to fix.

fullfatwasabi committed:
Small styling fixes

fullfatwasabi committed:
Small styling fixes

fullfatwasabi committed:
Add deletion confirmation box

User might accidentally delete task, and it may be hard to remember or notice what they deleted, and they might not have saved recently.

Let's add a confirmation popup

fullfatwasabi committed:
no message

fullfatwasabi committed:
Add assertions

Code has no checks to ensure save file is of the correct format.

This may be risky and might result in errors if the save file is improperly formatted eg corrupted.

Lets add assertions to ensure the data is validly formatted as it is loaded.

fullfatwasabi committed:
Amend code

Code has several areas where code quality such as method names are improperly implemented.

This might be messy and makes for harder readability

Let's rewrite some parts to follow closer to good code quality practices.

fullfatwasabi committed:
Merge pull request #1 from fullfatwasabi/branch-A-Assertions

Add assertions

fullfatwasabi committed:
Merge branch-A-CodeQuality

Merged Master branch that was pulled from PR request to branch-A-CodeQuality

fullfatwasabi committed:
Add "Continuous Integration" functionality

fullfatwasabi committed:
Merge branch 'branch-A-CodeQuality'

fullfatwasabi committed:
Add duplicate task detection.

Users might accidentally add tasks multiple times.

This would make things confusing and troublesome for them.

Add a feature that detects when the user adds an item that is identical to one that is in the list, and display a confirmation popup to ensure the user wishes to add another task identical to one already in the list.

fullfatwasabi committed:
Amend some code

fullfatwasabi committed:
Checkstyle fixes

fullfatwasabi committed:
Add some icons

Currently Sados buttons and app has no icons.

This can look quite boring and as if the app is incomplete.

Lets add some nice icons :)

fullfatwasabi committed:
Add a user guide

Currently no user guide is provided.

Users might not know how to use this marvelous life-changing app.

Lets add a quick and easy to use user guide.

fullfatwasabi committed:
Checkstyle fixes

fullfatwasabi committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	save.txt
#	src/main/java/addon/Storage.java
#	src/main/java/addon/Task.java
#	src/main/java/head/Sados.java

fullfatwasabi committed:
representative screenshot fix

fullfatwasabi committed:
Moving some files

fullfatwasabi committed:
Delete Ui.PNG

fullfatwasabi committed:
Ui.png extension fix

fullfatwasabi committed:
Set theme jekyll-theme-slate

fullfatwasabi committed:
Set theme jekyll-theme-slate

fullfatwasabi committed:
Readding Junit Tests

Seems like I accidentally removed the test.java for the junit testing, which is sad and stupid.

Lets readd the file back in :)

fullfatwasabi committed:
Merge remote-tracking branch 'origin/master'

fullfatwasabi committed:
Checkstyle changes

fullfatwasabi committed:
Fixed gradle errors

fullfatwasabi committed:
Fix font bug for popup window

fullfatwasabi committed:
Fix font bug for main program window

fullfatwasabi committed:
Fix compatibility issues.

fullfatwasabi committed:
Merge branch 'master' of https://github.com/fullfatwasabi/ip

fullfatwasabi committed:
Code cleanup

fullfatwasabi committed:
Bug fix

fullfatwasabi committed:
Checkstyle fixes
